### PR TITLE
[FEATURE] Expression variables relating to QGIS environment

### PR DESCRIPTION
**Pull request by nyalldawson**
New variables for:
- @qgis_os_name: eg 'linux','windows' or 'osx'
- @qgis_platform: eg 'desktop' or 'server'
- @user_account_name: current user's operating system account name
- @user_full_name: current user's name from os account (if available)

### CODE REVIEW

File: src/server/qgsserver.cpp [hunk 1]
```diff
@@ -331,6 +331,7 @@ bool QgsServer::init( int & argc, char ** argv )
   }
 
   mQgsApplication = new QgsApplication( argc, argv, getenv( "DISPLAY" ) );
+  mQgsApplication->sPlatformName = "server";
```
Comment (m-kuhn) [inline src/server/qgsserver.cpp]: No setter?

**Comment by nyalldawson** [comment]
Merged in 91f3005f806a75308dd4726272c2ac5f113858db

File: src/server/qgsserver.cpp (hunk 1 above)
Comment (nyalldawson) [inline src/server/qgsserver.cpp]: I moved it to the constructor instead

**Comment by SebDieBln** [comment]
> Merged in 91f3005

... but PR not closed? :wink:

**Comment by nyalldawson** [comment]
Feature freeze stress ;)

**Pull request closed without merging**